Master port to OM - AgentStats -rebased

s7design opened the pull request.

mgruner commented on Kernel/Modules/AgentStats.pm:
```diff
@@ -2819,20 +2844,23 @@ sub _ColumnAndRowTranslation {
+
     # create the needed language object
     use Kernel::Language;
     $Self->{LanguageObject} = Kernel::Language->new(
```
Here a local $LanguageObject might be better, no? Also passing in the objects will not have any effect there I believe, UserLanguage should be enough.

mgruner commented on scripts/test/Selenium/Agent/AgentStats/AgentStatsRun.t:
```diff
@@ -0,0 +1,252 @@
+        $Selenium->switch_to_window( $Handles->[0] );
+
+        # delete edited imported test stats
+        $Selenium->find_element("//a[contains(\@href, \'Subaction=Delete\' )]")->click();
```
I'm getting an error here in Selenium/FF because a print dialog opens in the subscreen. Then the main dialog asks for a confirmation if the page can be left. The modal dialog blocks the form submit, causing an error.

mgruner commented:
Hi S7, please see my inline comments.

I also get errors on AgentStatsOverview.t:

FailedTests:
  /ws/otrs5/scripts/test/Selenium/Agent/AgentStats/AgentStatsOverview.t #32 - StatID=1 - Title = 'Changes of status in a monthly overview' - found on screen, 34 - StatID=1 - Object = 'StateAction' - found on screen, 36 - StatID=1 - Description = 'Monthly overview, which reports status changes per day of a selected' - found on screen, 44 - StatID=2 - Title = 'List of tickets created last month' - found on screen, 48 - StatID=2 - Description = 'List of all tickets created last month. Order by age.' - found on screen, 56 - StatID=3 - Title = 'New Tickets' - found on screen, 58 - StatID=3 - Object = 'TicketAccumulation' - found on screen, 60 - StatID=3 - Description = 'Total number of new tickets per day and queue which have been created' - found on screen, 68 - StatID=4 - Title = 'List of the most time-consuming tickets' - found on screen, 72 - StatID=4 - Description = 'List of tickets closed last month which required the most time to' - found on screen, 80 - StatID=5 - Title = 'List of tickets closed, sorted by solution time' - found on screen, 84 - StatID=5 - Description = 'List of tickets closed last month, sorted by solution time.' - found on screen, 92 - StatID=6 - Title = 'List of open tickets, sorted by time left until response deadline expires' - found on screen, 96 - StatID=6 - Description = 'List of open tickets, sorted by time left until response deadline' - found on screen, 104 - StatID=7 - Title = 'Overview about all tickets in the system' - found on screen, 106 - StatID=7 - Object = 'TicketAccumulation' - found on screen, 108 - StatID=7 - Description = 'Current state of all tickets in the system without time restrictions.' - found on screen, 116 - StatID=8 - Title = 'List of open tickets, sorted by time left until escalation deadline expires' - found on screen, 120 - StatID=8 - Description = 'List of open tickets, sorted by time left until escalation deadline' - found on screen, 128 - StatID=9 - Title = 'List of tickets closed, sorted by response time.' - found on screen, 130 - StatID=9 - Object = 'Ticketlist' - found on screen, 132 - StatID=9 - Description = 'List of tickets closed last month, sorted by response time.' - found on screen, 134 - StatID=9 - Format = 'CSV' - found on screen, 140 - StatID=10 - Title = 'List of open tickets, sorted by time left until solution deadline expires' - found on screen, 142 - StatID=10 - Object = 'Ticketlist' - found on screen, 144 - StatID=10 - Description = 'List of open tickets, sorted by time left until solution deadline' - found on screen, 146 - StatID=10 - Format = 'CSV' - found on screen, 152 - StatID=11 - Title = 'List of tickets closed last month' - found on screen, 154 - StatID=11 - Object = 'Ticketlist' - found on screen, 156 - StatID=11 - Description = 'List of all tickets closed last month. Order by age.' - found on screen, 158 - StatID=11 - Format = 'CSV' - found on screen

s7design commented:
Hi @mgruner 

I cleaned code in the last commit. 

Regarding to:
* LangugaeObject you was right, we omitted this while we had ported.
* AgentStatsOverview.t : I think that it is works fine, but maybe your installation of OTRS default language was  German, and because of that German stats *.de.xml is loaded. I had such problem, and I instaled new OTRS without default language set during installation (i.e. English is default). Because of that we used English stats for testing stats overview. If you have any suggestion how we can test it on another way, please let me know.
* and for AgentStatsRun.t : I browsed about this, there is a problem generally with default  modal dialog (e.g. save dialog,  Print dialog ... ). Because of that I removed code for testing Delete stats, and added clean up stats data in xml_storage. I believe that it is important that there is cleaned system after testing. It is maybe not the best solution by then we will not find solution for modal dialog it is well.

Please let me know what do you think about all of that.

Regards
Zoran

mgruner commented:
Merged, thanks a lot! This concludes Kernel/Modules for OM porting AFAICT. I simplified the Overview test because I also had other problems (different order of stats). This should be ok.

s7design commented:
Yes you are right, all modules in Kernel/Modules are ported to OM now.

I will see changed Selenium test for Overview. We thought about the best test for stats overview, and it is a little strange, because I have not found the best test scenario for it. When I say this, I mean that there are different stats that depend on languages, and as you said with different stats order in overview. But the most important is that we finished the first phase of porting. We are continuing our work on others modules for porting :)

Regards
Zoran
